src/rest-server/src/controllers/v2/user.js

```diff
@@ -20,13 +20,25 @@ const userModel = require('@pai/models/v2/user');
 const createError = require('@pai/utils/error');
 const authConfig = require('@pai/config/authn');
 const groupModel = require('@pai/models/v2/group');
+const vcModel = require('@pai/models/vc');
+
+const getUserVCs = async (username) => {
+  const userInfo = await userModel.getUser(username);
+  let virtualClusters = new Set();
+  for (const group of userInfo.grouplist) {
+    const groupVCs = await groupModel.getGroupVCs(group);
+    virtualClusters = new Set([...virtualClusters, ...groupVCs]);
+  }
+  return [...virtualClusters];
+};
 
 const getUser = async (req, res, next) => {
   try {
     const username = req.params.username;
     const userInfo = await userModel.getUser(username);
-    userInfo['admin'] = userInfo.grouplist.includes(authConfig.groupConfig.adminGroup.groupname);
-    userInfo['virtualCluster'] = userInfo['extension']['virtualCluster'] ? userInfo['extension']['virtualCluster'] : [];
+    const groupItems = await groupModel.getListGroup(userInfo.grouplist);
+    userInfo['admin'] = groupModel.getAdminWithGroupinfo(groupItems);
+    userInfo['virtualCluster'] = await groupModel.getVCsWithGroupinfo(groupItems);
```

Comment:
getVCListWithGroupInfo will be much better

Reply:
Renamed getVCsWithGroupInfo